chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -987,6 +992,104 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
         Ok(block_index)
     }
 
+    pub fn get_stake_pool_balances_at_heights(
+        &self,
+        pool_ids: &[PoolId],
+        min_height: BlockHeight,
+        max_height: BlockHeight,
+    ) -> Result<BTreeMap<BlockHeight, BTreeMap<PoolId, Amount>>, BlockError> {
```

Comment:
Maybe use a range?

Reply:
> Maybe use a range?

Do you insist on this?
Rust's ranges are focused around iteration but it turns out we can't iterate over a range of `BlockHeight`s (the iteration requires to implement `std::iter::Step` which is not stable yet). And outside of iteration, `Range` doesn't seem to enforce any relation between its boundaries, so it just gives some false sense of safety IMO. Also, the naming of fields/methods is confusing - both `Range` and `RangeInclusive` have `end` and you really have to look at the type to understand whether its inclusive or exclusive (I'd prefer to give specific names to inclusive range boundaries, like "min/max" or "first/last", but not "start/end").

Reply:
> Maybe use a range?

[discussed] Leaving this as is.